File: src/components_cpp/src/node2.cpp


#include "components_cpp/node2.hpp"


Node2::Node2() : Node("node2")
{
    this->timer4_ = this->create_wall_timer(1000ms, std::bind(&Node2::callbackTimer4, this));
    this->timer5_ = this->create_wall_timer(1000ms, std::bind(&Node2::callbackTimer5, this));
}

void Node2::callbackTimer4()
{
    std::this_thread::sleep_for(2000ms);
    RCLCPP_INFO(this->get_logger(), "cb 4");
}

void Node2::callbackTimer5()
{
    std::this_thread::sleep_for(2000ms);
    RCLCPP_INFO(this->get_logger(), "cb 5");
}

File: src/components_py/components_py/node2.py
import rclpy
from rclpy.node import Node
import time
class Node2(Node):
    def __init__(self):
        super().__init__("node2")
        self.timer4_ = self.create_timer(1.0, self.callback_timer4)
        self.timer5_ = self.create_timer(1.0, self.callback_timer5)

    def callback_timer4(self):
        time.sleep(2.0)
        self.get_logger().info("cb 4")

    def callback_timer5(self):
        time.sleep(2.0)
        self.get_logger().info("cb 5")